packages/vm/src/traits.rs

```diff
@@ -27,7 +27,35 @@ impl<S: Storage, A: Api, Q: Querier> Extern<S, A, Q> {
 }
 
 #[cfg(feature = "iterator")]
-pub type StorageIteratorItem = FfiResult<(KV, u64)>;
+pub type NextItem = (Option<KV>, u64);
+
+#[cfg(feature = "iterator")]
+pub trait StorageIterator {
+    fn next(&mut self) -> FfiResult<NextItem>;
+
+    /// Collects all elements, ignoring gas costs
+    fn elements(mut self) -> FfiResult<Vec<KV>>
+    where
+        Self: Sized,
+    {
+        let mut out: Vec<KV> = Vec::new();
+        loop {
+            let (next, _gas_used) = self.next()?;
+            if let Some(kv) = next {
+                out.push(kv);
+            } else {
+                break;
+            }
+        }
+        Ok(out)
+    }
+}
+
+impl<I: StorageIterator + ?Sized> StorageIterator for Box<I> {
+    fn next(&mut self) -> FfiResult<NextItem> {
+        (**self).next()
+    }
+}
```

Comment:
This is based on this:
https://doc.rust-lang.org/std/boxed/struct.Box.html#impl-Iterator

Reply:
Thanks!